parlai/utils/testing.py

```diff
@@ -364,3 +364,96 @@ def display_model(opt) -> Tuple[str, str, str]:
         popt['datatype'] = 'test:stream'
         dm.display_model(popt)
     return (train_output.getvalue(), valid_output.getvalue(), test_output.getvalue())
+
+
+class AutoTeacherTest:
+    def _run_display_data(self, datatype, **kwargs):
+        import parlai.scripts.display_data as dd
+
+        dd.DisplayData.main(
+            task=self.task, datatype=datatype, display_verbose=True, **kwargs
+        )
+
+    def test_train(self):
+        """
+        Test --datatype train.
+        """
+        return self._run_display_data('train')
+
+    def test_train_stream(self):
```

Comment:
This especially is good to have, because I've had teachers that would specifically break in the streaming case...